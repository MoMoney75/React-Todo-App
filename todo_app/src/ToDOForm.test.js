import TodoForm from "./NewTodoForm";
import App from "./App";
import {render,fireEvent,container} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'

test("Render of form component", function(){
    render(<TodoForm />)
})

test("Snapshot test of form component", function(){
    const {asFragment} = render(<TodoForm />)
    expect(asFragment()).toMatchSnapshot()
})



test("Changing value of form input", function(){
const {getByLabelText,getByText,container} = render(<App />)

expect(container.querySelector("ol")).toBeInTheDocument()
expect(container.querySelector("li")).not.toBeInTheDocument()
expect(getByLabelText("todo:").value).toEqual("")

fireEvent.change(getByLabelText("todo:"),
{target :{name:"todo", value: "Eat more chickens"}})

expect(getByLabelText("todo:").value).toEqual("Eat more chickens")

expect(container.querySelector("li")).not.toBeInTheDocument()
fireEvent.submit(getByText("submit"))
expect(container.querySelector("li")).toBeInTheDocument()

})