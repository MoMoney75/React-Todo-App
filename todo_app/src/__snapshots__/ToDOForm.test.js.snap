// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of form component 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="todo"
      >
        todo:
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="enter task"
        type="text"
        value=""
      />
      <button>
        submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
